### PR TITLE
BadRequest validation added to Kamus API

### LINKED ISSUE #93: Kamus API should return 400 on invalid request
When the required properties are not present in encrypt or decrypt request, the api returns 500 instead of 400 bad request.

### PR DESCRIPTION
**Pull request by AleF83**
Close #93

### CODE REVIEW

**Comment by omerlh** [comment]
@AleF83 bumb version please

File: src/encrypt-api/Controllers/EncryptController.cs [hunk 1]
```diff
@@ -28,23 +29,28 @@ public EncryptController(IKeyManagement keyManagement)
         [Route("api/v1/encrypt")]
         public async Task<ActionResult> Encrypt([FromBody]EncryptRequest body)
         {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest("One of required fields doesn't present in request body.");
```
Comment (omerlh) [inline src/encrypt-api/Controllers/EncryptController.cs]: Log the validation error please

File: src/decrypt-api/Controllers/DecryptController.cs [hunk 2]
```diff
@@ -35,6 +35,11 @@ public DecryptController(IKubernetes kubernetes, IKeyManagement keyManagement)
         [Authorize(AuthenticationSchemes = "kubernetes")]
         public async Task<ActionResult> Decrypt([FromBody] DecryptRequest body)
         {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest("One of required fields doesn't present in request body.");
```
Comment (omerlh) [inline src/decrypt-api/Controllers/DecryptController.cs]: Log validation error please

**Comment by AleF83** [comment]
See last commit. Do you mean that?

**Comment by AleF83** [comment]
I want to add some tests to blackbox. Something that isn't happy flow.

File: src/encrypt-api/encrypt-api.csproj [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <TargetFramework>netcoreapp2.1</TargetFramework>
   </PropertyGroup>
   <PropertyGroup>
-    <Version>0.1.0.0</Version>
+    <Version>0.1.0.1</Version>
```
Comment (omerlh) [inline src/encrypt-api/encrypt-api.csproj]:
```suggestion
    <Version>0.1.1.0</Version>
```

File: src/decrypt-api/decrypt-api.csproj [hunk 4]
```diff
@@ -3,7 +3,7 @@
     <TargetFramework>netcoreapp2.1</TargetFramework>
   </PropertyGroup>
   <PropertyGroup>
-    <Version>0.1.0.0</Version>
+    <Version>0.1.0.1</Version>
```
Comment (omerlh) [inline src/decrypt-api/decrypt-api.csproj]:
```suggestion
    <Version>0.1.1.0</Version>
```

File: src/decrypt-api/Controllers/DecryptController.cs [hunk 5]
```diff
@@ -35,6 +35,14 @@ public DecryptController(IKubernetes kubernetes, IKeyManagement keyManagement)
         [Authorize(AuthenticationSchemes = "kubernetes")]
         public async Task<ActionResult> Decrypt([FromBody] DecryptRequest body)
         {
+            if (!ModelState.IsValid)
+            {
+                mAuditLogger.Warning("Bad request to Decrypt API: {validationState}",
```
Comment (omerlh) [inline src/decrypt-api/Controllers/DecryptController.cs]:
```suggestion
                mAuditLogger.Warning("Bad request to Decrypt API: {remoteIP} {validationState}",
```

File: src/decrypt-api/Controllers/DecryptController.cs [hunk 6]
```diff
@@ -35,6 +35,14 @@ public DecryptController(IKubernetes kubernetes, IKeyManagement keyManagement)
         [Authorize(AuthenticationSchemes = "kubernetes")]
         public async Task<ActionResult> Decrypt([FromBody] DecryptRequest body)
         {
+            if (!ModelState.IsValid)
+            {
+                mAuditLogger.Warning("Bad request to Decrypt API: {validationState}",
+                    Request.HttpContext.Connection.RemoteIpAddress,
+                    ModelState.ValidationState);
+                return BadRequest("One of required fields doesn't present in request body.");
```
Comment (omerlh) [inline src/decrypt-api/Controllers/DecryptController.cs]:
```suggestion
                return BadRequest("One or more of required fields doesn't present in the request body.");
```

File: src/encrypt-api/Controllers/EncryptController.cs [hunk 7]
```diff
@@ -28,23 +29,31 @@ public EncryptController(IKeyManagement keyManagement)
         [Route("api/v1/encrypt")]
         public async Task<ActionResult> Encrypt([FromBody]EncryptRequest body)
         {
+            if (!ModelState.IsValid)
+            {
+                mAuditLogger.Warning("Bad request to Encrypt API: {validationState}",
```
Comment (omerlh) [inline src/encrypt-api/Controllers/EncryptController.cs]:
```suggestion
                mAuditLogger.Warning("Bad request to Encrypt API: {sourceIP} {validationState}",
```

File: src/encrypt-api/Controllers/EncryptController.cs [hunk 8]
```diff
@@ -28,23 +29,31 @@ public EncryptController(IKeyManagement keyManagement)
         [Route("api/v1/encrypt")]
         public async Task<ActionResult> Encrypt([FromBody]EncryptRequest body)
         {
+            if (!ModelState.IsValid)
+            {
+                mAuditLogger.Warning("Bad request to Encrypt API: {validationState}",
+                    Request.HttpContext.Connection.RemoteIpAddress,
+                    ModelState.ValidationState);
+                return BadRequest("One of required fields doesn't present in request body.");
```
Comment (omerlh) [inline src/encrypt-api/Controllers/EncryptController.cs]:
```suggestion
                return BadRequest("One or more of the required fields doesn't present in the request body.");
```

File: tests/blackbox/EncryptControllerTests.cs [hunk 9]
```diff
@@ -14,22 +14,22 @@ namespace blackbox
...
-        [JsonProperty(PropertyName = "service-account", Required = Required.Always)]
+
+        [JsonProperty(PropertyName = "service-account")]
         public string SerivceAccountName
         {
             get;
             set;
         }
 
-        [JsonProperty(PropertyName = "namespace", Required = Required.Always)]
```
Comment (omerlh) [inline tests/blackbox/EncryptControllerTests.cs]: Why remove the required?

File: tests/blackbox/EncryptControllerTests.cs [hunk 10]
```diff
@@ -125,5 +125,22 @@ public async Task AnonymousRequestToDecryptEndpointShouldFail()
 
             Assert.Equal(HttpStatusCode.Unauthorized, result.StatusCode);
         }
+
+        [Fact]
+        public async Task BadRequestShouldFail()
```
Comment (omerlh) [inline tests/blackbox/EncryptControllerTests.cs]:
```suggestion
        public async Task Encrypt_BadRequest_ShouldFail()
```
Also, what about decrypt?

File: src/decrypt-api/Controllers/DecryptController.cs [hunk 11]
```diff
@@ -35,6 +35,14 @@ public DecryptController(IKubernetes kubernetes, IKeyManagement keyManagement)
         [Authorize(AuthenticationSchemes = "kubernetes")]
         public async Task<ActionResult> Decrypt([FromBody] DecryptRequest body)
         {
+            if (!ModelState.IsValid)
+            {
+                mAuditLogger.Warning("Bad request to Decrypt API: {remoteIP} {validationState}",
```
Comment (omerlh) [inline src/decrypt-api/Controllers/DecryptController.cs]:
```suggestion
                mAuditLogger.Warning("Bad request to Decrypt API: {sourceIP} {validationState}",
```

File: tests/blackbox/EncryptControllerTests.cs [hunk 12]
```diff
@@ -125,5 +125,22 @@ public async Task AnonymousRequestToDecryptEndpointShouldFail()
 
             Assert.Equal(HttpStatusCode.Unauthorized, result.StatusCode);
         }
+
+        [Fact]
+        public async Task Encrypt_BadRequest_ShouldFail()
```
Comment (omerlh) [inline tests/blackbox/EncryptControllerTests.cs]: What about decryptor tests?

**Pull request merged**